<launch>
    <arg name="map_frame" default="icp_map" doc="{icp_map,fused_map}"/>
    <arg name="control_odom_frame" default="odom"/>
    <arg name="robot_radius" default="$(eval 0.4)"/>
    <arg name="ground_truth_control" default="false"/>
    <arg name="paths_filename" default="$(dirname)/../../data/control/test_paths.csv"/>
    <arg name="paths_filename_gt" default="$(dirname)/../../data/control/test_paths_gt.csv"/>
    <arg name="results_filename" default="$(dirname)/../../data/control/results.txt"/>
    <arg name="max_path_deviation" default="$(eval 0.2)"/>
    <arg name="save_results" default="false"/>
    <arg name="run_control" default="true"/>
    <arg name="world" value="stage_4" />

    <group unless="$(arg ground_truth_control)">
        <node name="control_evaluator" pkg="aro_exploration" type="path_follower_evaluator.py" output="screen">
            <rosparam subst_value="true">
                robot_frame: base_footprint
                map_frame: $(arg map_frame)
                odom_frame: $(arg control_odom_frame)
                goal_reached_dist: $(arg robot_radius)
                paths_filename: $(arg paths_filename)
                results_filename: $(arg results_filename)
                max_path_deviation: $(arg max_path_deviation)
                save_results: $(arg save_results)
            </rosparam>
        </node>

        <include file="$(dirname)/aro_control_sim_dry.launch" pass_all_args="true"></include>
    </group>

    <group if="$(arg ground_truth_control)">
<!--         <node name="control_evaluator" pkg="aro_exploration" type="path_follower_evaluator.py" output="screen"> -->
<!--             <rosparam subst_value="true"> -->
<!--                 robot_frame: base_footprint -->
<!--                 map_frame: world -->
<!--                 odom_frame: world -->
<!--                 goal_reached_dist: $(arg robot_radius) -->
<!--                 paths_filename: $(arg paths_filename_gt) -->
<!--                 results_filename: $(arg results_filename) -->
<!--                 max_path_deviation: $(arg max_path_deviation) -->
<!--                 save_results: $(arg save_results) -->
<!--             </rosparam> -->
<!--         </node> -->

        <include file="$(dirname)/aro_control_sim_dry.launch" pass_all_args="true">
            <arg name="fixed_frame" value="world"/>
            <arg name="odom_frame" value="ground_truth_odom"/>
            <arg name="map_frame" value="world" />
            <arg name="control_odom_frame" value="world" />
        </include>
    </group>

</launch>
